django_loci/static/django-loci/js/loci.js

```diff
@@ -377,4 +464,90 @@ django.jQuery(function ($) {
         $indoor.show();
         $indoorRows.show();
     }
+
+    var geocode_api = window.location.origin + '/admin/django_loci/location/geocode/';
+    var reverse_api = window.location.origin + '/admin/django_loci/location/reverse-geocode/';
+    window.setTimeout(function () {
+        window.address_lat = null;
+        window.address_lng = null;
+        window.received_alert = false;
+        window.pressed_ok = false;
+        var marker_coords = get_marker() !== undefined ? get_marker().getLatLng() : null;
+        $addressInput.change(function () {
+            update_map(geocode_api);
+        });
+        var timeout = null;
+        $addressInput.keyup(function () {
+            window.clearTimeout(timeout);
+            timeout = window.setTimeout(function () {
+                if ($addressInput.val().length >= 10) {
+                    update_map(geocode_api);
+                }
+            }, 1000);
+        });
+
+        var submit = false;
+        $('form').submit(function (e) {
+            var current_marker_coords = get_marker() !== undefined ? get_marker().getLatLng() : null;
+            var current_marker_lat = get_marker() !== undefined ? get_marker().getLatLng().lat : null;
+            var current_marker_lng = get_marker() !== undefined ? get_marker().getLatLng().lng : null;
+            if (current_marker_coords !== marker_coords && current_marker_coords !== null) {
+                var reverse_api_with_data = reverse_api + '?lat=' + current_marker_lat + '&lng=' + current_marker_lng;
+                window.fetch(reverse_api_with_data)
+                    .then(function (resp) {
+                        var data_promise = resp.json();
+                        data_promise.then(function (data) {
+                            if ($addressInput.val()) {
+                                if (current_marker_lat !== window.address_lat ||
+                                        current_marker_lng !== window.address_lng) {
+                                    if (resp.status === 200) {
+                                        if (window.confirm('Would you like to update address?')) {
+                                            $addressInput.val(data.address);
+                                            submit = true;
+                                            $('form').submit();
+                                        } else {
+                                            submit = true;
+                                            $('form').submit();
+                                        }
+                                    } else {
+                                        var message = 'Current location does not have address, '
```

Comment:
I think in this case we can simply let the user know that we didn't find any address, something like:

```
alert("We couldn't find a related address to the location selected on the map, please enter the address manually")
```